### PR TITLE
Synonym Rules APIs docs

### PR DESCRIPTION
**Pull request by carlosdelest**
Follow up to https://github.com/elastic/elasticsearch/pull/98015. Adds Synonym Rules APIs, that allow individual synonym rules to be managed in a synonym set.

Added  `_cluster/stats` docs in this PR as well.

### CODE REVIEW

**Review by kderusso — APPROVED** [review]
Looks good! Some minor wording, pluralization nits and one question.

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (carlosdelest) [reply docs/reference/synonyms/apis/put-synonym-rule.asciidoc:144]: We don't - it's invalid when it does not conform to the Solr format. 

Do you think it's worth to explicitly name the format, and have another link here for it?

File: docs/reference/cluster/stats.asciidoc
Comment (carlosdelest) [reply docs/reference/cluster/stats.asciidoc:875]: I was thinking on that as well, but [token filters](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-tokenfilter.html#_solr_synonyms) refer to this as "synonym files".

I'm happy to change the wording to plurals for synonyms files as well, I'd just like to have docs team thoughts on this 👍

File: docs/reference/cluster/stats.asciidoc
Comment (carlosdelest) [reply docs/reference/cluster/stats.asciidoc:902]: I'm not so sure about these simplifications - I think it's useful for users to understand when things are counted as separate. Even if we err in the side of being too explicit, I think it's better to specify this clearly.

Any other thoughts from anyone?

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (kderusso) [reply docs/reference/synonyms/apis/put-synonym-rule.asciidoc]: Hmm. A quick Google search shows that our docs on validity aren't that comprehensive as-is. WDYT? Leave as is or try to expand?

File: docs/reference/cluster/stats.asciidoc
Comment (leemthompo) [reply docs/reference/cluster/stats.asciidoc:902]: I agree that explicit is fine here. 

BTW synonyms sets --> _synonym_ sets 😄

File: docs/reference/cluster/stats.asciidoc
Comment (leemthompo) [reply docs/reference/cluster/stats.asciidoc:884]: _synonym_ paths

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/synonyms-apis.asciidoc:25]:
```suggestion
=== Synonym sets APIs
```

this applies throughout

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/synonyms-apis.asciidoc:21]:
```suggestion
These filters are applied as part of the <<analysis,analysis>> process by the <<search-analyzer,search analyzer>>.
```

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/synonyms-apis.asciidoc:42]: I feel like this could be clearer...

Do we want to explain what format the rules come in? I understand that there are 2 flavors:

- equivalency (where terms are equal substitutes in the query)
- explicit mappings (where terms are mapped to one explicit term prior to querying)

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/synonyms-apis.asciidoc:43]:
```suggestion
You can manage synonym rules for a specific synonym set instead of updating the complete set.
```

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/put-synonym-rule.asciidoc:10]:
```suggestion
Creates or updates a synonym rule for a synonyms set.
```

File: docs/reference/synonyms/apis/delete-synonym-rule.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/delete-synonym-rule.asciidoc:135]:
```suggestion
If the synonym set is used in an analyzer, <<synonyms-set-analyzer-reloading,analyzer reloading>> is performed automatically.
```

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/put-synonym-rule.asciidoc:142]:
```suggestion
If the synonyms set is used in an analyzer, <<synonyms-set-analyzer-reloading,analyzer reloading>> is performed automatically.
```

File: docs/reference/synonyms/apis/put-synonyms-set.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/put-synonyms-set.asciidoc:117]: Again I'd recommend renaming the file/ids to use `synonym-set` for consistency

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (carlosdelest) [reply docs/reference/synonyms/apis/synonyms-apis.asciidoc:42]: We can do that. I was thinking on providing a specific synonyms guide that would explain the format (as per this proposal) and other places would link to it.

Take into account that we're describing the format in both the [synonym](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-tokenfilter.html) and [synonym graph](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-graph-tokenfilter.html) token filter. It would be nice to do so in a single place IMO.

File: docs/reference/cluster/stats.asciidoc
Comment (leemthompo) [reply docs/reference/cluster/stats.asciidoc:902]: Ah sorry, `synonyms set(s)` just reads a bit weird... 🙇 but feel free to ignore that point. I see @chriscressman recommended pluralization but I'm not sure if he meant synonym(s) or set(s) 🙈. Apologies for this nitty weirdness...

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (leemthompo) [reply docs/reference/synonyms/apis/synonyms-apis.asciidoc:42]: Sounds good

File: docs/reference/cluster/stats.asciidoc
Comment (mayya-sharipova) [inline docs/reference/cluster/stats.asciidoc:845]: The top `synonyms` section contains information about all `synonyms` not only "inline". We need to remove "inline" here, like this:

Contains statistics about synonyms defined  in <<analysis-synonym-tokenfilter,synonym>> and <<analysis-synonym-graph-tokenfilter,synonym graph>> token filters configuration.

---
"synonyms defined as `synonyms`" this sound strange, let's drop the second "synonyms"

File: docs/reference/cluster/stats.asciidoc
Comment (mayya-sharipova) [reply docs/reference/cluster/stats.asciidoc:902]: I like @kderusso's correction, I think it is simpler to understand. But I would leave this to you.

I would just write:

"Occurrences of unique synonyms sets in selected nodes."

File: docs/reference/synonyms/apis/delete-synonym-rule.asciidoc
Comment (mayya-sharipova) [reply docs/reference/synonyms/apis/delete-synonym-rule.asciidoc:135]: I would rephrase this without "if" statement, which seems to be little confusing. Like this:

"All analyzers using this synonyms set will reloaded automatically to reflect the new rules"

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (mayya-sharipova) [reply docs/reference/synonyms/apis/put-synonym-rule.asciidoc:142]: Here as well, I think this looks more understandable:

"All analyzers using this synonyms set will reloaded automatically to reflect the updated rule"

File: docs/reference/synonyms/apis/synonyms-apis.asciidoc
Comment (mayya-sharipova) [reply docs/reference/synonyms/apis/synonyms-apis.asciidoc:43]: How about this modification:
You can manage synonym rules => You can update individual synonym rules

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks for the great documentation. Overall LGTM, I left minor comments, which most are optional and suggestions, except the comment in stats where we need to remove "inline"  wording.

File: docs/reference/cluster/stats.asciidoc
Comment (carlosdelest) [reply docs/reference/cluster/stats.asciidoc:845]: After discussing with the team, I'm proposing some changes to the stats in https://github.com/elastic/enterprise-search-team/issues/5508. Happy to discuss there, I'll change the docs once that PR is merged.

File: docs/reference/cluster/stats.asciidoc
Comment (carlosdelest) [reply docs/reference/cluster/stats.asciidoc:902]: I like `Occurrences of unique synonyms sets`, changing that

File: docs/reference/synonyms/apis/delete-synonym-rule.asciidoc
Comment (carlosdelest) [reply docs/reference/synonyms/apis/delete-synonym-rule.asciidoc:135]: Good rephrasing @mayya-sharipova , adding that

File: docs/reference/synonyms/apis/put-synonym-rule.asciidoc
Comment (carlosdelest) [reply docs/reference/synonyms/apis/put-synonym-rule.asciidoc:142]: 👍

File: docs/reference/cluster/stats.asciidoc
Comment (carlosdelest) [reply docs/reference/cluster/stats.asciidoc:845]: Other comments are 🔝 , adding them

File: docs/reference/synonyms/apis/delete-synonym-rule.asciidoc
Comment (carlosdelest) [reply docs/reference/synonyms/apis/delete-synonym-rule.asciidoc]: Yes, already done 👍

**Comment by leemthompo** [comment]
Got some broken docs links per CI:

```12:40:29 INFO:build_docs:Bad cross-document links:
12:40:29 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/client/javascript-api/master/api-reference.html contains broken links to:
12:40:29 INFO:build_docs:   - en/elasticsearch/reference/master/delete-synonyms.html
12:40:29 INFO:build_docs:   - en/elasticsearch/reference/master/get-synonyms.html
12:40:29 INFO:build_docs:   - en/elasticsearch/reference/master/put-synonyms.html```

File: docs/reference/synonyms/apis/put-synonyms-set.asciidoc
Comment (leemthompo) [inline docs/reference/synonyms/apis/put-synonyms-set.asciidoc:47]:
```suggestion
(Required, string)
```

**Review by leemthompo — APPROVED** [review]
LGTM once broken links situation is resolved

**Comment by carlosdelest** [comment]
> Looks like this is due to the redirects deletion? cc @abdonpijpelink

It was! I've updated the redirects and now it passes. Thanks for the hint!

**Pull request merged**
